code/xcvm/lib/core/src/escrow.rs

```diff
@@ -51,51 +58,18 @@ pub enum ExecuteMsg {
 #[cfg_attr(feature = "std", derive(schemars::JsonSchema))]
 pub enum QueryMsg {}
 
-/// Deposits assets onto the virtual wallet.
-///
-/// This triggers a message to the accounts contract on the Centauri chain
-/// which updates user balances.  Because this is a cross-chain operation,
-/// there is a delay between this operation succeeding and funds showing up
-/// on user account.
+/// Message attached to [`::cw20::Cw20ReceiveMsg`] sent when receiving CW20
+/// funds.
 ///
 /// If the `account` doesn’t exist on the accounts contract, the deposit is
 /// aborted and assets are returned to the sender of the message.
+#[cfg(feature = "cw20")]
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
 #[serde(rename_all = "snake_case")]
 #[cfg_attr(feature = "std", derive(schemars::JsonSchema))]
-pub struct DepositAssetsRequest {
+pub struct ReceiveMsgBody {
 	/// Name of the account in the virtual wallet to deposit funds to.
 	pub account: String,
-	/// Funds attached to this message to deposit to the user.
-	pub deposits: Vec<LocalAssetAmount>,
-}
-
-/// An asset with its amount.
-#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
-#[serde(rename_all = "snake_case")]
-#[cfg_attr(feature = "std", derive(schemars::JsonSchema))]
-pub struct LocalAssetAmount {
-	/// Local asset identifier.
-	pub asset_id: LocalAssetId,
-	/// Amount of the asset.
-	pub amount: u128,
-}
-
-/// Local asset identifier.  XXX TODO
-#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
-#[serde(rename_all = "snake_case")]
-#[cfg_attr(feature = "std", derive(schemars::JsonSchema))]
-pub enum LocalAssetId {
-	Native { denom: String },
```

Comment:
asset is always native. always. i mean has local denomination. but also it  has external path if it was bridges. more like struct

Reply:
There can be either funds attached to the message and those have denomination or there can be CW20 assets which have contract addresses.  Is that not the case?

Reply:
you are right.

Reply:
in terms of cw20 assets to be moved. sure. but for remote assets not sure.